Folder	-  Read1 	Read2
DvS003Glioma1RNA-A-day3	- V300009963_L3_DKRT190110OligoTB19-19_1.fq.gz, V300009963_L3_DKRT190110OligoTB19-19_2.fq.gz
DvS003Glioma1RNA-B-day3	- V300009963_L3_DKRT190110OligoTB20-20_1.fq.gz,	V300009963_L3_DKRT190110OligoTB20-20_2.fq.gz 
DvS003Glioma1RNA-C-day3	- V300009963_L3_DKRT190110OligoTB21-21_1.fq.gz,	V300009963_L3_DKRT190110OligoTB21-21_2.fq.gz
DvS003Glioma1RNA-chops1	- V300009963_L4_DKRT190110OligoTB27-27_1.fq.gz,	V300009963_L4_DKRT190110OligoTB27-27_2.fq.gz
DvS003Glioma1RNA-chops2	- V300009963_L4_DKRT190110OligoTB28-28_1.fq.gz,	V300009963_L4_DKRT190110OligoTB28-28_2.fq.gz
DvS003Glioma1RNA-chops3	- V300009963_L4_DKRT190110OligoTB29-29_1.fq.gz,	V300009963_L4_DKRT190110OligoTB29-29_2.fq.gz
DvS003Glioma1RNA-D-day10	- V300009963_L4_DKRT190110OligoTB25-25_1.fq.gz, 300009963_L4_DKRT190110OligoTB25-25_2.fq.gz
DvS003Glioma1RNA-D-day3	- V300009963_L3_DKRT190110OligoTB22-22_1.fq.gz,	V300009963_L3_DKRT190110OligoTB22-22_2.fq.gz
DvS003Glioma1RNA-E-day10	- 300009963_L4_DKRT190110OligoTB26-26_1.fq.gz,	300009963_L4_DKRT190110OligoTB26-26_2.fq.gz
DvS003Glioma1RNA-G-day3	- 300009963_L3_DKRT190110OligoTB23-23_1.fq.gz,	00009963_L3_DKRT190110OligoTB23-23_2.fq.gz
DvS003Glioma1RNA-H-day3	- 300009963_L3_DKRT190110OligoTB24-24_1.fq.gz,	00009963_L3_DKRT190110OligoTB24-24_2.fq.gz

Downloading sequences from BGI- genomics
cd /work/users/ankushs/
mkdir Miao_glioblastoma
cd Miao_glioblastoma

#DvS003Glioma1RNA-A-day3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB19-19_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB19-19_2.fq.gz

#DvS003Glioma1RNA-B-day3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-B-day3/V300009963_L3_DKRT190110OligoTB20-20_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-B-day3/V300009963_L3_DKRT190110OligoTB20-20_2.fq.gz

#DvS003Glioma1RNA-C-day3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-C-day3/V300009963_L3_DKRT190110OligoTB21-21_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-C-day3/V300009963_L3_DKRT190110OligoTB21-21_2.fq.gz

#DvS003Glioma1RNA-chops1
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-chops1/V300009963_L4_DKRT190110OligoTB27-27_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-chops1/V300009963_L4_DKRT190110OligoTB27-27_2.fq.gz

#DvS003Glioma1RNA-chops2
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-chops2/V300009963_L4_DKRT190110OligoTB28-28_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-chops2/V300009963_L4_DKRT190110OligoTB28-28_2.fq.gz

#DvS003Glioma1RNA-chops3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-chops3/V300009963_L4_DKRT190110OligoTB29-29_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-chops3/V300009963_L4_DKRT190110OligoTB29-29_2.fq.gz

#DvS003Glioma1RNA-D-day10
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-D-day10/V300009963_L4_DKRT190110OligoTB25-25_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-D-day10/V300009963_L4_DKRT190110OligoTB25-25_2.fq.gz

#DvS003Glioma1RNA-D-day3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-D-day3/V300009963_L3_DKRT190110OligoTB22-22_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-D-day3/V300009963_L3_DKRT190110OligoTB22-22_2.fq.gz

#DvS003Glioma1RNA-E-day10
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-E-day10/V300009963_L4_DKRT190110OligoTB26-26_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-E-day10/V300009963_L4_DKRT190110OligoTB26-26_2.fq.gz

#DvS003Glioma1RNA-G-day3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-G-day3/V300009963_L3_DKRT190110OligoTB23-23_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-G-day3/V300009963_L3_DKRT190110OligoTB23-23_2.fq.gz

#DvS003Glioma1RNA-H-day3
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-H-day3/V300009963_L3_DKRT190110OligoTB24-24_1.fq.gz
wget --user=F18FTSEUHT1779_HUMhgnR --password='ekL3b4' ftp://5.57.48.133/Clean/DvS003Glioma1RNA-H-day3/V300009963_L3_DKRT190110OligoTB24-24_2.fq.gz


#Path to files 
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB19-19_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB20-20_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB21-21_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB27-27_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB28-28_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB29-29_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB25-25_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB22-22_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB26-26_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB23-23_1.fq.gz
/work/users/ankushs/Miao_glioblastoma/300009963_L3_DKRT190110OligoTB24-24_1.fq.gz

/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB19-19_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB20-20_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB21-21_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB27-27_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB28-28_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB29-29_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB25-25_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB22-22_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L4_DKRT190110OligoTB26-26_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/V300009963_L3_DKRT190110OligoTB23-23_2.fq.gz
/work/users/ankushs/Miao_glioblastoma/300009963_L3_DKRT190110OligoTB24-24_2.fq.gz

###shell script for alignment using TOPHAT2 to reference genome hg38

#!/bin/bash
#SBATCH --job-name=Glioma1_tophat2
# Project:
#SBATCH --account=
# Wall clock limit:
#SBATCH --time=48:00:00
# Max memory usage:
#SBATCH --mem-per-cpu=6000
# Number of cores:
#SBATCH --cpus-per-task=8
#SBATCH --mail-user=ankush.sharma@ibv.uio.no
#SBATCH --mail-type=FAIL
#SBATCH --mail-type=END

#If you need hugemem...
#--partition=hugemem


## Set up the job environment
source /cluster/bin/jobsetup
module purge
module load openmpi.intel
module load tophat
module load bowtie2
module load samtools
module load python3

transcriptome_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/bowtie2-index/trans_th2_idx 
ensembl_genome_ref=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/bowtie2-index/ 
read_path=work/users/ankushs/Miao_glioblastoma/
GTF_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf 
set -o errexit

# -r and --mate-std-dev parameters are determined using picard-tool "CollectInsertSizeMetrics"

# Aligning SEQUENCING DATA TO REFERENCE GENOME  :: TOPHAT_script.sh
mkdir log

# -r and --mate-std-dev parameters are determined using picard-tool "CollectInsertSizeMetrics"

# Aligning SEQUENCING DATA TO REFERENCE GENOME  :: TOPHAT_script.sh

#tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o pEF1_U2992_3_NonPhiX_tophat2 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/pEF1_U2992_3_NonPhiX_R1.fq $read_path/pEF1_U2992_3_NonPhiX_R2.fq 2> log/pEF1_U2992_3_NonPhiX_tophat2.out2 

tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-A-day3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB19-19_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB19-19_2.fq.gz 2> log/DvS003Glioma1RNA-A-day3.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-B-day3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB20-20_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB20-20_2.fq.gz 2> log/DvS003Glioma1RNA-B-day3.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-C-day3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB21-21_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB21-21_2.fq.gz 2> log/DvS003Glioma1RNA-C-day3.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-chops1 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB27-27_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB27-27_2.fq.gz 2> log/DvS003Glioma1RNA-chops1.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-chops2 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB19-28_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB19-28_2.fq.gz 2> log/DvS003Glioma1RNA-chops2.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-chops3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB29-29_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB29-29_2.fq.gz 2> log/DvS003Glioma1RNA-chops3.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-D-day10 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB25-25_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB25-25_2.fq.gz 2> log/DvS003Glioma1RNA-D-day10.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-D-day3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB22-22_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB22-22_2.fq.gz 2> log/DDvS003Glioma1RNA-D-day3.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-E-day10 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB26-26_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB26-26_2.fq.gz 2> log/DvS003Glioma1RNA-E-day10.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-G-day3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB23-23_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB23-23_2.fq.gz 2> log/DvS003Glioma1RNA-G-day3.out
tophat2 -p 5 -r -23 --mate-std-dev 83 --library-type fr-firststrand --no-mixed --no-novel-juncs --transcriptome-index $transcriptome_path/known -o DvS003Glioma1RNA-H-day3 $ensembl_genome_ref/Homo_sapiens.GRCh38.dna.primary_assembly $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB24-24_1.fq.gz $read_path/DvS003Glioma1RNA-A-day3/V300009963_L3_DKRT190110OligoTB24-24_2.fq.gz 2> log/DvS003Glioma1RNA-H-day3.out

echo tophat2 alignment successful



####SORTING AND HTCOUNTING together 



!/bin/bash

#SBATCH --job-name=htcount1
# Project:
#SBATCH --account=NN9632K
# Wall clock limit:
#SBATCH --time=18:00:00
#SBATCH --output=/work/users/ankushs/software/htseq/htseq-%j.out
#SBATCH --error=/work/users/ankushs/software/htseq/htseq-%j.err
# Number of cores:
# Max memory usage:
#SBATCH --mem-per-cpu=6000
# Number of cores:
#SBATCH --cpus-per-task=8
#SBATCH --mail-user=ankush.sharma@ibv.uio.no
#SBATCH --mail-type=FAIL
#SBATCH --mail-type=END



#If you need hugemem...
#--partition=hugemem



## Set up the job environment
source /cluster/bin/jobsetup
module purge
module load samtools
module load Python/3.5.2-foss-2016b

#transcriptome_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/bowtie2-index/trans_th2_idx 
#ensembl_genome_ref=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/bowtie2-index/ 
#read_path=/work/users/ankushs/Miao_glioblastoma/
set -o errexit

samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops1/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > /chops1/htcount_DvS003Glioma1RNA-chops1.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops2/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-chops2.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-chops3.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-D-day10/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-D-day10.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-E-day10/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-E-day10.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-A-day3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-A-day3.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-B-day3.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-C-day3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-C-day3.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-D-day3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-D-day3.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-G-day3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-G-day3.txt
samtools sort -on /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-H-day3/accepted_hits.bam | samtools view - | htseq-count -i gene_id -f bam --additional-attr=gene_name - /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf > htcount_DvS003Glioma1RNA-H-day3.txt




SORTING and HTCOUNTING SEPARATELY

samtool sorting 

#!/bin/bash
#SBATCH --job-name=sort_samtools
# Project:
#SBATCH --account=NN9632K
# Wall clock limit:
#SBATCH --time=12:00:00
# Max memory usage:
#SBATCH --mem-per-cpu=3000MB
# Number of cores:
#SBATCH --cpus-per-task=12
#SBATCH --mail-user=ankush.sharma@ibv.uio.no
#SBATCH --mail-type=FAIL
#SBATCH --mail-type=END

#If you need hugemem...
#--partition=hugemem


## Set up the job environment
source /cluster/bin/jobsetup
module purge
module load samtools 
set -o errexit

#samtools sort -n accepted_hits.bam accepted_hits_sorted.bam
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops1/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops1/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops2/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops2/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops3/accepted_hits_sorted.bam 
samtools sort -n  /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-D-day10/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-D-day10/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-E-day10/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-E-day10/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-A-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-A-day3/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-C-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-C-day3/accepted_hits_sorted.bam 
samtools sort -n  /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-D-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-D-day3/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-G-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-G-day3/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-H-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-H-day3/accepted_hits_sorted.bam 
samtools sort -n /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits.bam -o /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits_sorted.bam





###HT COUNT SEQ
#!/bin/bash

#SBATCH --job-name=htcount1
# Project:
#SBATCH --account=NN9632K
# Wall clock limit:
#SBATCH --time=18:00:00
#SBATCH --output=/work/users/ankushs/software/htseq/htseq-%j.out
#SBATCH --error=/work/users/ankushs/software/htseq/htseq-%j.err
# Number of cores:
# Max memory usage:
#SBATCH --mem-per-cpu=6000
# Number of cores:
#SBATCH --cpus-per-task=8
#SBATCH --mail-user=ankush.sharma@ibv.uio.no
#SBATCH --mail-type=FAIL
#SBATCH --mail-type=END



#If you need hugemem...
#--partition=hugemem
RUNPATH=/work/users/ankushs/software/htseq/
cd $RUNPATH

source /work/users/ankushs/software/htseq/htseq/bin/activate


#transcriptome_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/bowtie2-index/trans_th2_idx
#ensembl_genome_ref=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/bowtie2-index/
#read_path=work/users/ankushs/Miao_glioblastoma/
GTF_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/





htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops1/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-chops1.txt
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops2/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-chops2.txt
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops_DvS003Glioma1RNA-chops3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-chops3.txt
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-D-day10/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-D-day10.txt
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day10_DvS003Glioma1RNA-E-day10/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-E-day10.txt
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-A-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-A-day3.txt
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-B-day3
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-C-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-C-day3
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-D-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-D-day3
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-G-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-G-day3
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-H-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-H-day3
htseq-count -i gene_id -f bam --additional-attr=gene_name /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits_sorted.bam /usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/Homo_sapiens.GRCh38.93.gtf -o /work/users/ankushs/Miao_glioblastoma/3_cuffdiff/HTCOUNT/DvS003Glioma1RNA-B-day3



####DIFFERENTIAL EXPRESSION USING CUFF DIFFF



#!/bin/bash
#SBATCH --job-name=K562_RNAseq_cuffdiff
# Project:
#SBATCH --account=nn9374k
# Wall clock limit:
#SBATCH --time=48:00:00
# Max memory usage:
#SBATCH --mem-per-cpu=4000MB
# Number of cores:
#SBATCH --cpus-per-task=8
#SBATCH --mail-user=ankush.sharma@ibv.uio.no
#SBATCH --mail-type=FAIL
#SBATCH --mail-type=END

#If you need hugemem...
#--partition=hugemem


## Set up the job environment
source /cluster/bin/jobsetup
module purge
module load cufflinks 
set -o errexit

genome_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/genome 
GTF_path=/usit/abel/u1/ankushs/nobackup/references/ensembl_build93/ensemble/gtf/ 

cuffdiff -p 18 -o . --library-type fr-firststrand -b $genome_path/Homo_sapiens.GRCh38.dna.primary_assembly.fa -u $GTF_path/Homo_sapiens.GRCh38.93.gtf -L D152V_U2992,hcM_U2992,pEF1,pEF1_U2992 ../2_tophat/D152V_U2992_1_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/D152V_U2992_2_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/D152V_U2992_3_NonPhiX_tophat2/accepted_hits.bam ../2_tophat/hcM_U2992_1_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/hcM_U2992_2_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/hcM_U2992_3_NonPhiX_tophat2/accepted_hits.bam ../2_tophat/pEF1_1_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/pEF1_2_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/pEF1_3_NonPhiX_tophat2/accepted_hits.bam ../2_tophat/pEF1_U2992_1_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/pEF1_U2992_2_NonPhiX_tophat2/accepted_hits.bam,../2_tophat/pEF1_U2992_3_NonPhiX_tophat2/accepted_hits.bam 2> cuffdiff.log2


cuffdiff -p 12 -o . --library-type fr-firststrand -b $genome_path/Homo_sapiens.GRCh38.dna.primary_assembly.fa -u $GTF_path/Homo_sapiens.GRCh38.93.gtf -L Chops1_DvS003Glioma1RNA-chops1,Chops2_DvS003Glioma1RNA-chops2,Chops3_DvS003Glioma1RNA-chops3,day10_DvS003Glioma1RNA-D-day10,day10E_DvS003Glioma1RNA-E-day10,day3_DvS003Glioma1RNA-D-day3,day3_DvS003Glioma1RNA-G-day3,day3_DvS003Glioma1RNA-G-day3,day3_DvS003Glioma1RNA-H-day3,
day3_DvS003Glioma1RNA-A-day3,day3_DvS003Glioma1RNA-B-day3,day3_DvS003Glioma1RNA-C-day3 /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops2_DvS003Glioma1RNA-chops2/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/Chops3_DvS003Glioma1RNA-chops3/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day10D_DvS003Glioma1RNA-D-day10/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day10E_DvS003Glioma1RNA-E-day10/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-D-day3/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-G-day3/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-H-day3/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-A-day3/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-B-day3/accepted_hits.bam /work/users/ankushs/Miao_glioblastoma/2_tophat/day3_DvS003Glioma1RNA-C-day3/accepted_hits.bam 2> cuffdiff.log2



